app/src/main/java/org/mozilla/rocket/periodic/PeriodicReceiver.kt

```diff
@@ -0,0 +1,57 @@
+package org.mozilla.rocket.periodic
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import androidx.work.OneTimeWorkRequest
+import androidx.work.WorkInfo
+import androidx.work.WorkManager
+import java.util.concurrent.TimeUnit
+
+class PeriodicReceiver : BroadcastReceiver() {
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        if(context == null) {
+            return
+        }
+        scheduleFirstLaunchWorker(context, WorkManager.getInstance())
+    }
+
+    private fun scheduleFirstLaunchWorker(context: Context, workManager: WorkManager) {
+        if(FirstLaunchWorker.isNotificationFired(context)){
+            return
+        }
+
+        val works = workManager.getWorkInfosByTag(FirstLaunchWorker.TAG).get()
+        works.forEach {
+            if(it?.state == WorkInfo.State.ENQUEUED){
+                /** Job already scheduled */
+                return
+            }
+        }
+
+
+        /** This hours can configure by firebase */
+        var delayHoursToInstallTime = 24
+
+        /** Find next scheduled hours */
+        val pm = context.packageManager
+        var firstInstallTime : Long = Long.MAX_VALUE
+        val packageInfo = pm.getPackageInfo(context.packageName, 0)
+        if(packageInfo != null && packageInfo.packageName == context.packageName){
```

Comment:
Wondering if we have any specific reason to do `packageInfo.packageName == context.packageName` sanity check